File: JavaScript/problem1.js
function problem1(max){
    var num_index = [];
    var sum = 0;

    for(var i = 1; i < max; i++){
        if((i%3 === 0) || (i%5 === 0)){
            console.log(i)
            num_index.push(i)
        }
    }

    for(var i = 0; i < num_index.length; i++){
        sum += num_index[i];
    }
    return sum;
}

console.log(problem1(1000));

File: Ruby/problem1.rb
def problem1(max)
  i = 1
  num_index = []
  while(i < max)
    if (i%3 == 0) || (i%5 == 0)
      num_index.push(i)
    end
    i += 1
  end

  sum = 0
  i = 0
  while i < num_index.count
    sum += num_index[i]
    i += 1
  end

  return sum
end

puts "problem 1 solution: " + problem1(1000).to_s
